assets/js/Containers/FrevaGPT/index.js

```diff
@@ -0,0 +1,438 @@
+
+class FrevaGPT extends React.Component {
+  // const abortController = useRef();
+  // abortController.current = new AbortController();
+  // const signal = abortController.signal;
+
+  constructor(props) {
+    super(props);
+    this.handleUserInput = this.handleUserInput.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.createNewChat = this.createNewChat.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.toggleBotSelect = this.toggleBotSelect.bind(this);
+    this.handleStop = this.handleStop.bind(this);
+    this.submitUserInput = this.submitUserInput.bind(this);
+
+    this.state = {
+      loading: false,
+      userInput: "",
+      botModelList: [],
+      botModel: "",
+      hideBotModelList: true,
+      botOkay: undefined,
+      showSuggestions: true,
+      dynamicAnswer: "",
+      dynamicVariant: "",
+    };
+  }
+
+  async componentDidMount() {
+    // if thread giving on mounting the component, set thread within store
+    const givenQueryParams = browserHistory.getCurrentLocation().query;
+    if (
+      has(givenQueryParams, "thread_id") &&
+      !isEmpty(givenQueryParams.thread_id)
+    ) {
+      this.props.dispatch(setThread(givenQueryParams.thread_id));
+
+      // request content of old thread if threa_id is given
+      this.setState({ loading: true });
+      await this.getOldThread(givenQueryParams.thread_id);
+      this.setState({ loading: false, showSuggestions: false });
+    }
+
+    const successfulPing = async () => {
+      let pingSuccessful = false;
+
+      try {
+        const response = await fetch("/api/chatbot/ping");
+        if (response.status === 200) pingSuccessful = true;
+      } catch (err) {
+        console.error("PingError: ", err);
+      }
+
+      return pingSuccessful;
+    };
+
+    const getBotModels = async () => {
+      const queryObject = {
+        auth_key: process.env.BOT_AUTH_KEY,
+      };
+      const response = await fetch(
+        `/api/chatbot/availablechatbots?` + objectToQueryString(queryObject)
+      );
+      this.setState({ botModelList: await response.json() });
+    };
+
+    if (await successfulPing()) {
+      this.setState({ botOkay: true });
+      await getBotModels();
+    } else this.setState({ botOkay: false });
+  }
+
+  createNewChat() {
+    this.props.dispatch(setConversation([]));
+    this.props.dispatch(setThread(""));
+    browserHistory.push({
+      pathname: "/chatbot/",
+      search: "",
+    });
+    this.setState({ showSuggestions: true });
+    window.scrollTo(0, 0);
+  }
+
+  handleUserInput(e) {
+    this.setState({ userInput: e.target.value });
+  }
+
+  async handleKeyDown(e) {
+    if (e.key === "Enter") {
+      this.handleSubmit(e.target.value);
+    }
+  }
+
+  async submitUserInput() {
+    const userInput = this.state.userInput;
+    await this.handleSubmit(userInput);
+    this.setState({ userInput: "" });
+  }
+
+  async handleSubmit(input) {
```

Comment:
It would be reasonable to prevent empty strings to be passed to the backend

Reply:
Adapted the affected functions. But edge cases like submitting a whitespace or similar absurd inputs are not covered. Should we thin kabout a more comprehensive approach? The Bot seems to handle this kind of inputs well (at least the one I tested).

Reply:
My take on this would be something like this as the first action in `handleSubmit` (since this is the function which performs the actual action either way)

```
    if (!input.trim()) {
      return;
    }
```

Or instead of negating the value (empty strings are falsy) you could as well surround it with `isEmpty`. By this we would at least cover whitespaces.